Simple-Projects/Simple_Projects/Basic/List/List.cpp

```diff
@@ -28,28 +27,23 @@ int List::max_size() const {
 }
 
 void List::erase(int index) {
-    if ( current > index && index > 0 ) {
-        current -= 1;
-
-        for (int i = 0; i < current; i++ ) {
-        	array[i] = array[i+1];
-        }
+    if ( current > index && index >= 0 ) {
+		std::copy(array + index + 1, array + current, array);
```

Comment:
Note the use of copy and copy_backward for menial copying - helps remove simple loops with copying.
Something to reflect on: why did I use copy_backward in the second case, not just copy?